### ISSUE TITLE
DB::Exception: Sizes of columns doesn't match

### ISSUE THREAD

**Issue by sergeyignatov** [yandex/ClickHouse#796]
Hi,
We have issue with ReplicatedGraphiteMergeTree. After 86400s (second retention) we are seeing lots of exceptions in logs. clickhouse version 1.1.54233

```SHOW CREATE TABLE graphite.graphite10

│ CREATE TABLE graphite.graphite10 ( Path String,  Value Float64,  Time UInt32,  Date Date,  Timestamp UInt32) ENGINE = ReplicatedGraphiteMergeTree(\'/clickhouse/tables/graphite10\', \'{replica}\', Date, (Path, Time), 8192, \'graphite_rollup10\') 

<graphite_rollup10>
  <default>
    <function>any</function>
    <retention>
      <age>0</age>
      <precision>10</precision>
    </retention>
    <retention>
      <age>86400</age>
      <precision>3600</precision>
    </retention>
  </default>
</graphite_rollup10>


2017.05.15 19:47:33.966779 [ 32 ] <Error> DB::StorageReplicatedMergeTree::queueTask()::<lambda(DB::StorageReplicatedMergeTree::LogEntryPtr&)>: Code: 9, e.displayText() = DB::Exception: Sizes of columns doesn't match: Path: 0, Value: 1, e.what() = DB::Exception, Stack trace:

0. clickhouse-server(StackTrace::StackTrace()+0x16) [0x299bb06]
1. clickhouse-server(DB::Exception::Exception(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, int)+0x1f) [0x10e177f]
2. clickhouse-server(DB::Block::checkNumberOfRows() const+0x263) [0x234c653]
3. clickhouse-server() [0x2b403f1]
4. clickhouse-server(DB::MergeTreeDataMerger::mergePartsToTemporaryPart(std::vector<std::shared_ptr<DB::MergeTreeDataPart const>, std::allocator<std::shared_ptr<DB::MergeTreeDataPart const> > >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, DB::MergeListEntry&, unsigned long, long, DB::DiskSpaceMonitor::Reservation*, bool)+0x2b2f) [0x2ae612f]
5. clickhouse-server(DB::StorageReplicatedMergeTree::executeLogEntry(DB::ReplicatedMergeTreeLogEntry const&)+0xd7c) [0x2ab13cc]
6. clickhouse-server() [0x2ab432d]
7. clickhouse-server(DB::ReplicatedMergeTreeQueue::processEntry(std::function<std::shared_ptr<zkutil::ZooKeeper> ()>, std::shared_ptr<DB::ReplicatedMergeTreeLogEntry>&, std::function<bool (std::shared_ptr<DB::ReplicatedMergeTreeLogEntry>&)>)+0x4a) [0x2b040ba]
8. clickhouse-server(DB::StorageReplicatedMergeTree::queueTask()+0x16c) [0x2a9455c]
9. clickhouse-server(DB::BackgroundProcessingPool::threadFunction()+0x454) [0x2afae64]
10. clickhouse-server() [0x3646630]
11. /lib/x86_64-linux-gnu/libpthread.so.0(+0x76ba) [0x7ffba03266ba]
12. /lib/x86_64-linux-gnu/libc.so.6(clone+0x6d) [0x7ffb9f94782d]
```

**Comment by ztlpn** [comment]
Hi @sergeyignatov ! This is a known bug, but we have not been able to pinpoint the cause of it yet. Here is how you can help: please provide all different strings of the form `Path: x, Value: y` that appear in the exception messages (I am interested in the values of x and y). Thanks!

**Comment by sergeyignatov** [comment]
grep -oP 'DB::Exception: Sizes of columns.*' /var/log/clickhouse-server/clickhouse-server.log |sort |uniq -c
  59457 DB::Exception: Sizes of columns doesn't match: Path: 0, Value: 1, e.what() = DB::Exception, Stack trace:

**Comment by ztlpn** [comment]
#810

**Comment by sergeyignatov** [comment]
with fix-graphite-rollup Exceptions are gone

**Comment by ztlpn** [comment]
fix released in v1.1.54236
